amalgamate.py

```diff
@@ -60,112 +68,152 @@
     "src/_decay.cpp",
 ]
 
+DEFAULT_FILES = [os.path.join("..", f) for f in DEFAULT_FILES]
+
+
+# Version Handling
+def get_version():
+    try:
+        version = (
+            subprocess.check_output(
+                ["git", "describe", "--tags"], stderr=subprocess.STDOUT
+            )
+            .strip()
+            .decode("utf-8")
+        )
+        if not version:
+            raise Exception("Empty version string from git.")
+        return version
+    except subprocess.CalledProcessError as e:
+        raise RuntimeError(
+            f"Git describe failed. Output: {e.output.decode('utf-8').strip()}\n"
+            "Hint: Ensure your repo has tags or fallback to .git_archival.txt."
+        )
+    except FileNotFoundError:
```

Comment:
It's not obvious how we get here?  Does the subprocess issue a `FileNotFoundError` exception?